cluster/get-kube-local.sh

```diff
@@ -66,103 +79,143 @@ function create_cluster {
   --pid=host \
   --privileged=true \
   -d \
-  gcr.io/google_containers/hyperkube-${arch}:${release} \
+  gcr.io/google_containers/hyperkube-${KUBE_ARCH}:${KUBE_VERSION} \
     /hyperkube kubelet \
       --containerized \
       --hostname-override="127.0.0.1" \
-      --address="0.0.0.0" \
       --api-servers=http://localhost:8080 \
       --config=/etc/kubernetes/manifests \
       --allow-privileged=true \
       --cluster-dns=10.0.0.10 \
       --cluster-domain=cluster.local \
       --v=2"
 
-  echo -e -n "\tWaiting for master components to start..."
-  while true; do
-    local running_count=$(kubectl -s=http://${KUBE_HOST}:8080 get pods --no-headers 2>/dev/null | grep "Running" | wc -l)
-    # We expect to have 3 running pods - etcd, master and kube-proxy.
-    if [ "$running_count" -ge 3 ]; then
-      break
-    fi
+  kube::log::indented "Waiting for master components to start..."
+
+  # We expect to have at least 3 running pods - etcd, master and kube-proxy.
+  while (($(${KUBECTL} get pods --no-headers 2>/dev/null | grep "Running" | wc -l) < 3)); do
     echo -n "."
-    sleep 1
+    sleep 2
   done
-  echo_green "SUCCESS"
-  echo_green "Cluster created!"
-  echo ""
-  kubectl -s http://${KUBE_HOST}:8080 clusterinfo
+
+  kube::log::green "SUCCESS"
+  kube::log::green "Cluster created!"
+}
+
+## UTILS
+
+# Installs kubectl for the specific os/arch
+kube::util::install_kubectl(){
+
+  kubectl_url="https://storage.googleapis.com/kubernetes-release/release/${KUBE_VERSION}/bin/${KUBE_PLATFORM}/kubectl"
+
+  if [[ ! -f $(which kubectl 2>&1) ]]; then
+    echo -n "Downloading kubectl binary to /usr/local/bin/kubectl..."
+    if [[ $(which wget) ]]; then
+      kube::util::run "wget -O /usr/local/bin/kubectl ${kubectl_url}"
+    elif [[ $(which curl) ]]; then
+      kube::util::run "curl -sSLO /usr/local/bin/kubectl ${kubectl_url}"
+    else
+      kube::log::red "Couldn't find curl or wget. Bailing out."
+      exit 1
+    fi
+    chmod a+x /usr/local/bin/kubectl
+  else
+    # TODO: We should detect version of kubectl binary if it too old and download newer version.
+    kube::log::normal "Detected existing kubectl binary. Skipping download."
+  fi
 }
 
-function get_latest_version_number {
+kube::util::get_latest_version() {
   local -r latest_url="https://storage.googleapis.com/kubernetes-release/release/stable.txt"
-  if [[ $(which wget) ]]; then
+  if [[ $(which wget 2>&1) ]]; then
     wget -qO- ${latest_url}
```

Comment:
Any reason not to use kube::util::run as per install_kubelet's example?

Given the common requirement between install_kubelet and get_latest_version, consider making a generic function for retrieving files via curl or wget (assuming this doesn't exist already somewhere in the kube repo).